### PR TITLE
Feature/epmdj 43 scopes no db

### PR DESCRIPTION
**Pull request by s0larm00n**

### CODE REVIEW

File: drill-common/src/commonMain/kotlin/com/epam/drill/common/MessageWithId.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.epam.drill.common
+
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class MessageWithId(
```
Comment (IgorKey) [inline drill-common/src/commonMain/kotlin/com/epam/drill/common/MessageWithId.kt]: Mmmm.... nice naming

File: drill-plugin-api/drill-admin-part/src/drillAdminPartMain/kotlin/com/epam/drill/plugin/api/end/AdminPluginPart.kt [hunk 2]
```diff
@@ -4,6 +4,12 @@ import com.epam.drill.common.AgentInfo
...
 
-abstract class AdminPluginPart(val sender: WsService, override val id: String) : DrillPlugin() {
+abstract class AdminPluginPart<A>(val sender: WsService, override val id: String) : DrillPlugin() {
+    abstract var actionSerializer: kotlinx.serialization.KSerializer<A>
+
     abstract suspend fun processData(agentInfo: AgentInfo, dm: DrillMessage): Any
+
+    abstract suspend fun doAction(agentInfo: AgentInfo, action: A)
+
+    abstract suspend fun doRawAction(agentInfo: AgentInfo, action: String)
```
Comment (IgorKey) [inline drill-plugin-api/drill-admin-part/src/drillAdminPartMain/kotlin/com/epam/drill/plugin/api/end/AdminPluginPart.kt]: It shouldn't be abstract. to move here

File: drill-admin/src/main/kotlin/com/epam/drill/endpoints/plugin/PluginDispatcher.kt [hunk 3]
```diff
@@ -4,6 +4,8 @@ package com.epam.drill.endpoints.plugin
 import com.epam.drill.common.AgentInfo
 import com.epam.drill.common.Message
 import com.epam.drill.common.MessageType
+import com.epam.drill.common.MessageWithId
+
```
Comment (IgorKey) [inline drill-admin/src/main/kotlin/com/epam/drill/endpoints/plugin/PluginDispatcher.kt]: why?

File: drill-admin/src/main/kotlin/com/epam/drill/endpoints/plugin/PluginDispatcher.kt [hunk 4]
```diff
@@ -84,23 +86,48 @@ class PluginDispatcher(override val kodein: Kodein) : KodeinAware {
...
-                        ?.send(
-                            Frame.Text(
-                                Message.serializer() stringify Message(
-                                    MessageType.MESSAGE,
-                                    "/plugins/action",
-                                    message
+                    val dp: DP? = plugins.plugins[ll.pluginId]
+                    val agentInfo = agentManager[ll.agentId]
+                    when {
+                        (dp == null) -> {
```
Comment (IgorKey) [inline drill-admin/src/main/kotlin/com/epam/drill/endpoints/plugin/PluginDispatcher.kt]: ugly code

File: drill-admin/src/main/kotlin/com/epam/drill/endpoints/plugin/PluginDispatcher.kt (hunk 3 above)
Comment (s0larm00n) [reply drill-admin/src/main/kotlin/com/epam/drill/endpoints/plugin/PluginDispatcher.kt]: do you mean the empty rоw or the MessageWithId import

**Review by IgorKey — APPROVED** [review]

**Pull request merged**